### PR TITLE
Gemcalib cami

### PR DESCRIPTION
**Pull request by cgalloni**

### CODE REVIEW

File: gemcalibration/include/gem/calib/Calibration.h [hunk 1]
```diff
@@ -75,26 +75,90 @@ namespace gem {
                 {"vfatChMin" , 0},
                 {"vfatChMax" , 0},
                 {"vt2"       , 0},
+		{"throttle"  ,100},
```
Comment (mexanick) [inline gemcalibration/include/gem/calib/Calibration.h]: please rename throttle-->trigThrottle

File: gemcalibration/include/gem/calib/Calibration.h [hunk 2]
```diff
@@ -75,26 +75,90 @@ namespace gem {
...
+                {"signalSourceType"       , 0},
                 }},
             {SCURVE,{
                 {"nSamples"  , 100},
-                {"trigType"  , 0},
+		{"trigType"  , 0}, // TODO: TTC local should be only possible one
```
Comment (mexanick) [inline gemcalibration/include/gem/calib/Calibration.h:83]: we will provide settings validation, this should not really touch the front-end, I think. A corresponding method should be called at the backend when the "APPLY" button is pressed and returning alert message should contain notification of possible errors (as it is done now for a check of OH mask). Eventually, we can extend these maps to a structures where we will have default and boundary values

File: gemcalibration/src/common/CalibrationWeb.cc [hunk 3]
```diff
@@ -110,16 +113,77 @@ void gem::calib::CalibrationWeb::comparatorSelector(xgi::Output* out)
...
+    *out << "<h2><span class=\"label label-warning col-md-6\">SIGNAL SOURCE"<< "</span></h2>"
+    << "<div class=\"col-md-6\">"
+        << "<div class=\"form-check\">"
+            << "<div class=\"form-check-input radio-inline\"> <label><input type=\"radio\" name=\"signalSourceType\" id=\"signalSource_radio_0\" value=0>Particle</label></div>"
+            << "<div class=\"form-check-input radio-inline\"> <label><input type=\"radio\" name=\"signalSourceType\" id=\"signalSource_radio_1\" value=1 checked>Calibration pulse</label></div>"
+        << "</div>"
+    << "</div>" << std::endl;    
+}
+
+void gem::calib::CalibrationWeb::dataTypeSelector(xgi::Output* out)//TODO should be set for Sbit Rate scan
```
Comment (mexanick) [inline gemcalibration/src/common/CalibrationWeb.cc]: I believe this is not needed as the outcoming data type is deduced from the calibration routine type and by design will be unique for each calibration routine. It is true, that we can perform certain scans using either tracking data, or internal FW counters, but the latter will be eliminated from current functionality.

File: gemcalibration/src/common/CalibrationWeb.cc [hunk 4]
```diff
@@ -110,16 +113,77 @@ void gem::calib::CalibrationWeb::comparatorSelector(xgi::Output* out)
...
+    *out << "<h2><span class=\"label label-warning col-md-6\">VFAT ADC reference"<< "</span></h2>"
+    << "<div class=\"col-md-6\">"
+        << "<div class=\"form-check\">"
+            << "<div class=\"form-check-input radio-inline\"> <label><input type=\"radio\" name=\"adcType\" id=\"adc_radio_0\" value=0 checked>Internal</label></div>"
+            << "<div class=\"form-check-input radio-inline\"> <label><input type=\"radio\" name=\"adcType\" id=\"adc_radio_1\" value=1>External</label></div>"
+        << "</div>"
+    << "</div>" << std::endl;    
+}
+
+void gem::calib::CalibrationWeb::tempSensorTypeSelector(xgi::Output* out)
```
Comment (mexanick) [inline gemcalibration/src/common/CalibrationWeb.cc]: We've concluded that the temperature monitoring is actually not a calibration routine, it does not calibrate anything. This has to be removed (and further implemented in the `gemdaqmonitor` subpackage)

File: gemcalibration/include/gem/calib/Calibration.h [hunk 5]
```diff
@@ -54,15 +54,113 @@ namespace gem {
...
+                {"latency"    , 33},
+		{"mspl"       , 4},
+		{"trimValues" , 0},// TODO: need to be implemented properly as taking array of numbers 
+                                   // TODO: need to implement interaction with DB to get proper configurations per ARM DAC
+		}},
+	    {DACSCANV3  ,{
+	       {"nSamples",0},
+	       {"adcType",0}
+	      }},// TODO: drop down with DACs to select to scan on, and a select all button
+	    {TEMPERATURE  ,{// TODO: Move to monitor page
```
Comment (mexanick) [inline gemcalibration/include/gem/calib/Calibration.h]: Update the list according to the comment about cal types

File: gemcalibration/include/gem/calib/Calibration.h [hunk 6]
```diff
@@ -54,15 +54,113 @@ namespace gem {
...
+	       {"trigType"  , 0}, // TODO: TTC local should be only possible one
+               {"l1aTime"   , 250},
+               {"pulseDelay", 40},
+               {"latency"   , 33},
+               {"armDacList" , 0},// TODO: need to be implemented properly as taking array of numbers 
+            
+               {"calPhase"  , 0}, 
+	       }}  
+        };
+	std::map< std::string,std::string > m_scanParamsLabels{
```
Comment (mexanick) [inline gemcalibration/include/gem/calib/Calibration.h:131]: I guess if we are going to introduce labels, we should think about merging all these maps into structures. 
Fine for the moment, but should be refactored later.

**Review by mexanick — COMMENTED** [review]
General: please check the settings of your editor w.r.t. indentations. (I believe correctly tuned editor should everything correctly without demanding your attention).  You can ask Jared for support as he uses emacs and knows how to tune it right. More detailed comments are inline. 
Also please change the base branch from `feature-helpCamilla` to a branch `feature-gemcalibration`. This branch will be deleted soon. I think you can to such change when editing PR.

Update: disregard the last one, I did it myself. Please issue future PRs to `feature-gemcalibration`

File: gemcalibration/include/gem/calib/Calibration.h (hunk 1 above)
Comment (cgalloni) [reply gemcalibration/include/gem/calib/Calibration.h]: OK, I'll do that.

File: gemcalibration/include/gem/calib/Calibration.h (hunk 2 above)
Comment (cgalloni) [reply gemcalibration/include/gem/calib/Calibration.h:83]: For this one. Wouldn't we want to remove completely the trigger option for the Scurve and just assign by default the local TTC option?

File: gemcalibration/src/common/CalibrationWeb.cc (hunk 3 above)
Comment (cgalloni) [reply gemcalibration/src/common/CalibrationWeb.cc]: Ok then I'll remove the option of the data type.

File: gemcalibration/src/common/CalibrationWeb.cc (hunk 4 above)
Comment (cgalloni) [reply gemcalibration/src/common/CalibrationWeb.cc]: yes, no problem, will remove it

File: gemcalibration/include/gem/calib/Calibration.h (hunk 5 above)
Comment (cgalloni) [reply gemcalibration/include/gem/calib/Calibration.h]: What do you mean here? 
I'll remove the TEMPERATURE. Would you like me to change the calType naming convention?

File: gemcalibration/include/gem/calib/Calibration.h (hunk 2 above)
Comment (mexanick) [reply gemcalibration/include/gem/calib/Calibration.h:83]: I think we leave it as is for the moment (with the default setting to TTC)... We can provide blocking later with the configuration tag, if needed.

File: gemcalibration/include/gem/calib/Calibration.h (hunk 5 above)
Comment (mexanick) [reply gemcalibration/include/gem/calib/Calibration.h]: yes, remove the TEMPERATURE and change the rest according to my comments at calType definition

File: gemcalibration/include/gem/calib/Calibration.h [hunk 7]
```diff
@@ -20,7 +20,7 @@
 namespace gem {
   namespace calib {
 
-    enum calType {NDEF, PHASE, LATENCY, SCURVE, SBITRATE, THRESHOLD, TRIMDAC, DACSCANV3, TEMPERATURE, SBITREADOUT, SBITMAPANDRATE}; 
+    enum calType {NDEF, PHASE, LATENCY, SCURVE, SBITRATE, THRESHOLD, TRIMDAC, DACSCANV3, TEMPERATURE, SBITREADOUT, SBITMAPANDRATE, CALIBRATEARMDAC};
```
Comment (mexanick) [inline gemcalibration/include/gem/calib/Calibration.h]: Ups, I guess my comment to this line got lost when I changed the base branch...
Please remove the TRESHOLD as it is the same as CALIBRATEARMDAC, remove the TEMPERATURE, SBITREADOUT and SBITMAPANDRATE as non-calibration tools. Rename the DACSCANV3 to ARMDACSCAN as per @dorney comment

Comment (cgalloni) [reply gemcalibration/include/gem/calib/Calibration.h]: Hello Misha, 
I removed the TEMPERATURE monitoring.
SBITREADOUT and SBITMAPANDRATE are still in there, but I can remove them right away if you confirm.  

For what concerns the  THRESHOLD, following Brian's comment, I changed it to ARMDACSCAN and looks like this atm.
Screen Shot 2019-03-28 at 4 39 07 PM
The CALIBRATEARMDAC in my understanding is another routine that had to be implemented and atm looks like, but should be modified  (Triggertype removed and list of ARM DAC settings to scan implemented)
Screen Shot 2019-03-28 at 4 22 33 PM
 DACSCANV3 is also a different one atm and looks like this:
Screen Shot 2019-03-28 at 4 56 52 PM

Please let me know if this is incorrect and which differences  are/aren't sopposed to be between these calibrations. 

Thanks and best, 
Camilla

**Comment by mexanick** [comment]
I think this looks correct now. Thank you!

**Review by mexanick — APPROVED** [review]

File: gemcalibration/include/gem/calib/Calibration.h (hunk 7 above)
Comment (cgalloni) [reply gemcalibration/include/gem/calib/Calibration.h]: Hi @mexanick , I removed also the SBITREADOUT and SBITMAPANDRATE, following your suggestion. Please mind that in my last PR the other Calibration routines discussed above are still there. 
Thanks and cheers,
Camilla

**Pull request merged**